sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -72,9 +74,71 @@ case class PartitionedFile(
   }
 }
 
+/**
+ * Class used to store statistical data that is collected during a file scan and could be used to
+ * update the SQL metrics of the scan node. More members could be added to this class to to collect
+ * metrics related to new features.
+ */
+case class FileScanMetrics(
+    topLevelVariantMetrics: Option[VariantMetrics] = None,
+    nestedVariantMetrics: Option[VariantMetrics] = None) {
+
+  // Update SQL metrics with the data in topLevelVariantMetrics and nestedVariantMetrics
+  def updateSQLMetrics(sqlMetrics: Option[Map[String, SQLMetric]]): Unit = {
+    sqlMetrics match {
+      case Some(sqlMetrics) =>
+        topLevelVariantMetrics match {
```

Comment:
I feel like `VariantConstructionMetrics` should have another method like `updateSQLMetrics`, which takes in a `Option[VariantMetrics]` and a `Map[String, SQLMetric]`, and does this update there.

Then, here, we could just call that method to do this update work. We might need to distinguish between nested and top-level, but that is the general idea.